### PR TITLE
Gh42   dead for negative bloodied

### LINKED ISSUE #42: "dead" for -bloodied
If you get a huge hit and knock the other monster to below negative 1/2 maxHP, they should be dead irretrievably.

This should be rare enough to almost never happen.

For level 1 monsters, the most they can hit for it about 14 on a natural 20. Most level 1 monsters maxHP is at least 24. In order for this to happen to a level 1 monster, they would have to have only 1HP left, and have their opponent roll at natural 20 to hit, and have their opponent have bonuses to hit.

### PR DESCRIPTION
**Pull request by cormacmccarthy**
fixes #42

### CODE REVIEW

File: creatures/base.js [hunk 1]
```diff
@@ -286,10 +304,6 @@ Battles won: ${this.battles.wins}`;
 		const hp = this.hp - damage;
 		const originalHP = this.hp;
 
-		if (hp <= 0) {
-			this.hp = 0;
-		}
-
```
Comment (tweettypography) [inline creatures/base.js]: Why this logic even in here??

File: creatures/base.js [hunk 2]
```diff
@@ -352,9 +366,17 @@ Battles won: ${this.battles.wins}`;
 	}
 
 	die (assailant) {
-		this.emit('die', {
-			assailant
-		});
+		if (this.hp <= 0) {
+			if (this.destroyed()){
```
Comment (tweettypography) [inline creatures/base.js]: Should be `this.destroyed` (not a function)

File: creatures/base.js [hunk 3]
```diff
@@ -352,9 +366,17 @@ Battles won: ${this.battles.wins}`;
...
+				this.emit('destroy', {
+					assailant
+				});
```
Comment (tweettypography) [inline creatures/base.js]: You might want to just issue the same `die` even and include the parameter `destroyed` - would probably make it simpler to handle all of the cases where the die event might be listened to...

File: ring/index.js [hunk 4]
```diff
@@ -29,8 +29,7 @@ class Ring extends BaseClass {
 			ring.battles.push(results);
 		});
 
-		const ring = this;
-		this.channelManager.on('win', (className, channel, { contestant }) => ring.handleWinner({ contestant }));
+		this.channelManager.on('win', (className, channel, { contestant }) => this.handleWinner({ contestant }));
 		this.channelManager.on('loss', (className, channel, { contestant }) => this.handleLoser({ contestant }));
 		this.channelManager.on('draw', (className, channel, { contestant }) => this.handleTied({ contestant }));
```
Comment (tweettypography) [inline ring/index.js]: I don't see you actually listening for the `permaDeath` event here?

File: creatures/base.js (hunk 2 above)
Comment (cormacmccarthy) [reply creatures/base.js]: oh weird... That's going to take some getting used to.

File: game.js [hunk 5]
```diff
@@ -185,14 +186,21 @@ ${monsterCard(monster, contestant.lastMonsterPlayed !== monster)}`
...
 		const channel = this.publicChannel;
+		let announce;
 
-		channel({
-			announce:
-`${monster.identityWithHp} is killed by ${assailant.identityWithHp}
+		if (destroyed) {
+			announce = `${monster.identityWithHp} has been sent to the land of ${monster.pronouns} fathers by ${assailant.identityWithHp}
+
+			☠️  R.I.P ${monster.identity}
```
Comment (tweettypography) [inline game.js]: Did you mean for there to be this extra whitespace on the left?

File: ring/index.js [hunk 6]
```diff
@@ -453,12 +453,21 @@ class Ring extends BaseClass {
 
 				if (contestant.monster.dead) {
 					contestant.lost = true;
-					this.channelManager.queueMessage({
-						announce: `${contestant.monster.givenName} has died in battle. You may now \`revive\` or \`dismiss\` ${contestant.monster.pronouns[1]}.`,
-						channel,
-						channelName,
-						event: { name: 'loss', properties: { contestant } }
-					});
+					if (contestant.monster.destroyed()) {
```
Comment (tweettypography) [inline ring/index.js]: I still think this should be `if (contestant.monster.destroyed) {` - I'm kind of confused how it's working...

File: game.js (hunk 5 above)
Comment (cormacmccarthy) [reply game.js]: Not originally, but I kinda like it. It stands out. I mean, it's pretty huge if this happens. I expect it to happen maybe 1 time ever TBH. (so much work for something so unlikely 🙄 )

**Comment by cormacmccarthy** [comment]
image

**Pull request merged**